File: scalang/src/main/scala/com/kapilkoju/autopos/config/LoggingAspectConfiguration.scala
package com.kapilkoju.autopos.config

import com.kapilkoju.autopos.aop.logging.LoggingAspect
import io.github.jhipster.config.JHipsterConstants
import org.springframework.context.annotation._
import org.springframework.core.env.Environment

@Configuration
@EnableAspectJAutoProxy class LoggingAspectConfiguration {

  @Bean
  @Profile(Array(JHipsterConstants.SPRING_PROFILE_DEVELOPMENT))
  def loggingAspect(env: Environment): LoggingAspect = {
    new LoggingAspect(env)
  }
}

File: src/main/kotlin/com/kapilkoju/autopos/config/LoggingAspectConfiguration.kt
package com.kapilkoju.autopos.config

import com.kapilkoju.autopos.aop.logging.LoggingAspect
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.EnableAspectJAutoProxy
import org.springframework.context.annotation.Profile
import org.springframework.core.env.Environment

@Configuration
@EnableAspectJAutoProxy
class LoggingAspectConfiguration {

    @Bean
    @Profile(value = [Profiles.DEVELOPMENT])
    fun loggingAspect(env: Environment): LoggingAspect {
        return LoggingAspect(env)
    }
}